src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java

```diff
@@ -202,13 +217,66 @@ public FSItem retrieveById(FSContainerInterface container,
         return retrieveWorker(firestore, datasetId, enumerateDepth, fireStoreDirectoryEntry, throwOnNotFound, fileId);
     }
 
+    /**
+     * Retrieve a batch of FSFile by id
+     *
+     * @param container - dataset or snapshot containing file's directory entry
+     * @param fileIds   - list of ids of  file identifiers - directory identifiers will throw
+     * @return list of FSItem of retrieved files; throws on not found
+     */
+    public List<FSFile> batchRetrieveById(
+        FSContainerInterface container,
+        List<String> fileIds,
+        int enumerateDepth,
+        boolean throwOnNotFound) throws InterruptedException {
+
+        Firestore firestore = FireStoreProject.get(getProjectIdForFSContainer(container)).getFirestore();
+        String containerId = container.getId().toString();
+
+        List<FireStoreDirectoryEntry> directoryEntries =
+            directoryDao.batchRetrieveById(firestore, containerId, fileIds);
+
+        // TODO: When we have more than one dataset in a snapshot then we will have to
+        //  split entries by underlying dataset. For now we know that they all come from one dataset.
+        List<FireStoreFile> files =
+            fileDao.batchRetrieveFileMetadata(firestore, containerId, directoryEntries);
+
+        List<FSFile> resultList = new ArrayList<>();
+        assert (directoryEntries.size() == files.size());
```

Comment:
what error does this throw if false?

Reply:
Actually i should not use it. I will replace it with a check and a throw. If assert checking is enabled, it will terminate the program.